### PR TITLE
feature/eap-fast-integration

### PR DESCRIPTION
**Pull request by jrouzierinverse**
## Description

Integrates eap-fast into packetfence using freeradius 3.1.x
## Impacts

Radius Workflow and Admin GUI
## NEWS file entries

New Features
++++++++++++
- EAP-FAST 
## Delete branch after merge

NO

### CODE REVIEW

File: conf/documentation.conf [hunk 1]
```diff
@@ -1875,3 +1875,25 @@ options=enabled|disabled
 description=<<EOT
 Show the parking portal to the device instead of the usual portal
 EOT
+
+[radius_authentication_methods.eap_authentication_types]
+type=multi
+options=GTC|MD5|MSCHAPv2|PEAP|FAST|LEAP
```
Comment (extrafu) [inline conf/documentation.conf]: Not sure why we list LEAP here - it is crap. How about TLS? TTLS?

Comment (jrouzierinverse) [inline conf/documentation.conf]: Should we also add SIM or not needed?

Comment (extrafu) [inline conf/documentation.conf]: I think we should eventually. IIRC there are stuff to configure for doing EAP-SIM that are outside of the scope of PacketFence. Let's add EAP-SIM in there when we actually handle this.

Comment (jrouzierinverse) [inline conf/documentation.conf]: So remove LEAP completely or keep it there since it is configured in 
freeradius?

James Rouzier
jrouzier@inverse.ca :: +1.514.447.4918 (x115)  ::  http://www.inverse.ca
Inverse inc. :: Leaders behind SOGo (http://www.sogo.nu) and PacketFence (http://www.packetfence.org)

File: conf/radiusd/packetfence-tunnel.example [hunk 2]
```diff
@@ -224,6 +224,9 @@ session {
 #  the ONLY safe way is to set "use_tunneled_reply = yes", and
 #  then update the inner-tunnel reply.
 post-auth {
+    update {
```
Comment (fdurand) [inline conf/radiusd/packetfence-tunnel.example]:
```
    update {
            &request: := &outer.request:[*]
    }
```

File: conf/radiusd/packetfence-tunnel.example [hunk 3]
```diff
@@ -234,6 +237,9 @@ post-auth {
 		-sql_reject
 	}
 
+	update {
```
Comment (fdurand) [inline conf/radiusd/packetfence-tunnel.example:241]: remove

File: conf/radiusd/packetfence.example [hunk 4]
```diff
@@ -322,13 +247,13 @@ post-auth {
...
-	#	&reply: += &session-state:
-	#}
+	update {
+		&reply: += &session-state:
+	}
 
 	# If the request has not been handled inside the inner-tunnel
 	# send it to packetfence
-	if (! EAP-Type || (EAP-Type != TTLS  && EAP-Type != PEAP) ) {
+	if (! EAP-Type || (EAP-Type != TTLS  && EAP-Type != PEAP && EAP-Type != FAST) ) {
```
Comment (fdurand) [inline conf/radiusd/packetfence.example]: if (! EAP-Type || (EAP-Type != TTLS  && EAP-Type != PEAP ) ) {

File: conf/radiusd/packetfence-tunnel.example [hunk 5]
```diff
@@ -224,6 +224,9 @@ session {
 #  the ONLY safe way is to set "use_tunneled_reply = yes", and
 #  then update the inner-tunnel reply.
 post-auth {
+    update {
+        &request: += &outer.request:[*]
+    }
 	#
 	rest
```
Comment (fdurand) [inline conf/radiusd/packetfence-tunnel.example:231]:
```
    if (EAP-Type != FAST) ) {
            rest
    }
```

**Comment by louismunro** [comment]
please rebase.

**Comment by extrafu** [comment]
I think it's rather https://github.com/inverse-inc/packetfence/pull/1700 we need to merge.

**Comment by louismunro** [comment]
Right.
This one should be closed if it's not useful.
